### DISCUSSION TITLE
Content scripts are not run in dev mode until they are changed

### DISCUSSION THREAD

**Discussion by renataogarcia** [wxt-dev/wxt#1248] [Q&A]
I'm aware of WTX [register them dynamically](https://wxt.dev/guide/resources/faq.html#why-aren-t-content-scripts-added-to-the-manifest), but is it only registering once there is a change to the script? Is this the expected behaviour?

**Comment by aklinker1** [comment]
It registers them as soon as the extension connects to the dev server. If you open the background's console and enable verbose logs, you should be able to see logs every time the content scripts are registered. The initial log says "reloaded", but that's because it uses the same function the first time as subsequent times, and I was too lazy to customize the message based on the number of times the function to load content scripts was called.


https://github.com/user-attachments/assets/20d3c853-236f-4d96-9caa-eba882cbc238

**Reply by aklinker1** [reply]
If you're seeing different behavior, open an issue with a reproduction so I can debug your case further.

**Reply by renataogarcia** [reply]
Thanks, I created https://github.com/wxt-dev/wxt/issues/1252
